### ISSUE TITLE
Feature Request: Would be great to be able to enumerate the priorities that are different! 

### ISSUE THREAD

**Issue by Simonl9l** [diegofrata/Generator.Equals#50]
Would be great to have a wrapper class such that one can have a Comparer collect the difference between say two `record`'s or other supported type. This would also walk the nested collection and say use a `.`, `[x]` or `=>` notation for the hierarchy, array/set elements or maps. 

A Use Case: [CDC](https://en.wikipedia.org/wiki/Change_data_capture) you have the before and after image from your database of choice and need detect what changes happen between them, so some conditional biz logic can be applied based upon that.

That this generator avoids reflection means it will also work with AoT compilation.

**Comment by diegofrata** [comment]
I haven't got much time recently to invest in new features. I will keep this here in case someone has a view on how this would be implemented/used.

**Comment by Nickztar** [comment]
I'd love to attempt this. Would be a very nice feature, would help when debugging aswell 👍🏼 

Will get back when I have a working example/PR.

**Comment by hayer** [comment]
How do you show the difference between `string`? Do a Myers diff?
What about `byte[]` or `string[]`? And is `char[]` treated as a `string`?
Then you have `DateOnly`, what is the result there? And the list goes on.

I love the idea but two things come to my mind;
 * This is very opinionated and quickly gets tied to business rules and logic.
 * It can quickly grow complex it probably deserves its own package, `Generator.Diffs`

**Comment by diegofrata** [comment]
Perhaps it is the wording, but my understanding is that this is a feature request to enumerate inequalities but not necessarily create diffs. I think that would be a reasonable feature to have, although I totally agree diffs are out of questions.

**Comment by vmachacek** [comment]
Yes I also would want to use this generator to generate log of changes effectively. Comparison is nice, but my use case requires to figure out what changed. I can use == and if that fails I can compare JSON but that is painfully ineffective
